spec/component_spec.rb

```diff
@@ -22,11 +22,15 @@ module Clearwater
     it 'sanitizes element attributes' do
       attributes = Component.sanitize_attributes({
         class: 'foo',
+        data_toggle: 'dropdown',
         onclick: proc { |event| expect(event).to be_a Bowser::Event },
       })
 
-      # Renames :class to :class_name
-      expect(attributes[:class_name]).to eq 'foo'
+      # Renames :class to className
+      expect(attributes['className']).to eq 'foo'
+
+      # move none standard html attributes to attributes[:attributes] and 'data_*' to 'data-*'
+      expect(attributes[:attributes]['data-toggle']).to eq 'dropdown'
```

Comment:
We use the JS DOM API to set DOM attributes, they're sometimes different from the HTML version. I don't think this is a bad thing. It'll be familiar to anyone who has worked with a virtual DOM before, like React. So `data-*` attributes are actually passed as a hash to the `dataset` attribute:

```ruby
  div(dataset: { foo: 'bar' })
```

This would result in `<div data-foo="bar"></div>`.

I didn't know you could pass in a nested `attributes` hash, though. That's pretty cool. Must be a `virtual-dom` feature. Unfortunately, it's a major change from how `Clearwater::Component` currently works. I think working almost directly with the JS DOM API is pretty powerful. I think trying to make it more HTML-like would make it feel less like a virtual DOM.